Packs/ShiftManagement/Scripts/GetOnCallHoursPerUser/GetOnCallHoursPerUser.py

```diff
@@ -0,0 +1,60 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+from typing import List, Dict
+
+
+def count_hours_in_shift(shift: Dict) -> int:
+    from_day = shift.get('fromDay', 0)
+    to_day = shift.get('toDay', 0)
+    from_hour = shift.get('fromHour', 0)
+    to_hour = shift.get('toHour', 0)
+    hours_in_shift = 0
+    for day in range(from_day, to_day + 1):
+        if day == from_day:
+            if day == to_day:
+                hours_in_shift += (to_hour - from_hour)
+            else:
+                hours_in_shift += (24 - from_hour)
+        elif day == to_day:
+            hours_in_shift += to_hour
+        else:
+            hours_in_shift += 24
+    return hours_in_shift
+
+
+def main():
+    get_roles_response: List = demisto.executeCommand('getRoles', {})
+    if is_error(get_roles_response):
+        demisto.error(f'Failed to get roles: {str(get_error(get_roles_response))}')
+    else:
+        hours_per_user: Dict[str, int] = {}
+        roles = get_roles_response[0]['Contents']
+        for role in roles:
+            get_users_response: List = demisto.executeCommand('getUsers', {'roles': role.get('name')})
```

Comment:
Should check if there are shifts for this role and only then retrieve uses. We might have 1000s of roles from AD, etc. and this can be extremely slow.

Reply:
But then we might miss users, and we want to display all users, including ones with no on-call hours
We can change the script to first get all users, initialize them all with 0 on-call hours and then query for users who are assigned to roles with shifts as you suggested
You think that will be better?

Reply:
Yes, sounds like a good suggestion. In that case, you might not even need to call `getUsers` for a role because you already have the data in your hands from the first `getUsers` call...

Reply:
done